### DISCUSSION TITLE
Beta or Stable release discussion

### DISCUSSION THREAD

**Discussion by stax76** [staxrip/staxrip#579] [General]
I can release Beta and Stable builds when @Dendraspis or contributors request or suggest it in this discussion thread.

Or @Dendraspis you just release it when you think it's time. If you also want to take care of Beta build releases also just release them if you have storage to upload it.

Otherwise, I can continue making releases based on requests and suggestions here.

**Comment by jessielw** [comment]
I've been using latest BETA. It's so far been pretty stable for me.

**Comment by Dendraspis** [comment]
I'm already thinking about a system, that is going to work for me, too. There are also a few question related to this topic.  😁 

The next release should be a Stable, but I think not before next week.

**Reply by stax76** [reply]
Sounds good.

**Comment by jessielw** [comment]
Will the new beta builds be released on Github or still via onedrive/dropbox?

**Reply by Dendraspis** [reply]
Depends on what we decide.  😁

**Comment by Dendraspis** [comment]
Regarding the progress since `v2.1.8.0` I suggest releasing **`v2.2.0.0 Stable`** and skipping  `v2.1.9.0`.

I also suggest changing the versioning. At the moment each number is **one** digit only for no reason.
In order to be able to release multiple hotfixes or even dev builds between beta versions, I suggest using the last digit for those builds. At the same time the digits in the middle should also be able to get higher numbers than `9`.

Examples:

- Next Stable: `v2.2.0.0`
- Following Betas: `2.2.1.0`, `2.2.2.0`, ..., `2.2.13.0`, `2.2.14.0`, `2.2.15.0`, ...
- Following Stable: `2.3.0.0`
- Following Betas: `2.3.1.0`, `2.3.2.0`, ..., `2.3.13.0`, ...

So someday we end up with something like this:
`v2.12.11.9` : 9th dev build of 11th beta that was based on `v2.12`

**Reply by JJKylee** [reply]
I totally agree. It looks more systematic. 👍

And a related suggestion:
I think all the beta versions should also be put in the GitHub `releases` archive along with the stable versions. And because the release terms for dev builds must be really short, we better put them in a personal cloud archive as @stax76 is currently doing for beta versions.

**Reply by ghost** [reply]
### Regarding BETA

Please do one more beta, because the folder structure change will CERTAINLY show some bugs. It would be harmful to release a stable with such a flaw. 

I can see already a problem, it's with the Authors.html.

@Stax76 I suspect there is a problem if the path has spaces (regarding Authors.html).
Also I think  you should take a look at the Authors.html page:
You should put : 
Coders and Designers: `Stax76 and Dendraspis`
Contributors: `Patman86, DJAtom and all Tools Authors`
Beta Testers: `JKyle and 44vince44`
Credits to authors before 2020: (..........)

We must do a beta, test it 3-4 days with NO NEW FEATURES ADDITIONS. If all ok, we do a new stable.

### Regarding version number

I agree with @Dendraspis, next stable will introduce major changes, so it should be a 2.2.0.0.

About the beta naming scheme, it doesn't have to be strict... I prefer slow number progression 😁 (imho)

**Comment by stax76** [comment]
@Dendraspis 

Is that semantic versioning compatible? If you want to change something, maybe it would make sense using semantic versioning, GitHub recommends it:

https://github.com/staxrip/staxrip/releases/new

https://semver.org

I don't know how it works and how popular or useful it is.

The release script checks FilePrivatePart to determine if it's Beta or Stable:

https://github.com/staxrip/staxrip/blob/master/Source/Release.ps1#L63

I can add some comments in case you want to use it as is or write your own based on it. I'm not sure if it still works after the folder move.

A few things are important, like not releasing a debug build of frameserver.dll, it would not run without Visual Studio being installed:

https://github.com/staxrip/staxrip/blob/master/Source/Release.ps1#L56

> I think all the beta versions should also be put in the GitHub releases archive along with the stable versions.

That would be cumbersome unless it can be scripted.

> I suspect there is a problem if the path has spaces (regarding Authors.html)

That failed indeed, it's now fixed.

> Also I think you should take a look at the Authors.html page

Currently, it shows only who is active or inactive, roles could be added similar to how mpc-be does it. Many projects don't maintain such a list I believe.

**Reply by Dendraspis** [reply]
This is a semantic versioning, like the one from the example, extended to 4 numbers to differ between stable and beta.  

The script should be still okay, because all stables end with `.0.0`, but let's see...

**Reply by ghost** [reply]
@stax76 

> Currently, it shows only who is active or inactive, roles could be added similar to how mpc-be does it. Many projects don't maintain such a list I believe.


Well then like in any credits, coders come first, beta-testers come last !

**Comment by Dendraspis** [comment]
I decided to re-think the versioning of StaxRip a few days ago, so I changed a few things and decided this:

Whenever there is a problem with an old build, the first step is to say:  *please test the latest beta, it might be fixed already.*  We have a lot of users, that use v2.1.8.0 or even v2.1.3.0, because they think it is the latest version they should use or that is is not old, because it's the second last version. The name **Beta** suggests a *not really ready for production usage* build and may annoy some people. Additionally we all know, that the released *Stable* versions are always a bit buggy.  😉 
So I decided to get rid of beta versions at all and only use one type of StaxRip.

The next build will be **v2.2.0**, followed by **v2.3.0**, **v2.4.0**, .. , **v2.19.0**, **v2.20.0**, ...
At the same time we should try to lower the frequency how often StaxRip is published. All these releases are published on GitHub.

In between there will be **DEV** builds like **v2.1.11**, **v2.2.3**, **v2.3.8**, ...
These DEV-build are "hotfixes", that are meant to fix urgent issues and/or to be used to let others test the fix. If there is no break included it is possible to release only the Executable on GitHub as a replacement.


I already updated the *Check for Updates* and in the next days I finish the script, that prepares the releases. So I hope you (basically) agree with my decision.  👼 

@stax76 
Please, don't beat me too hard.  👼 😁

**Reply by JJKylee** [reply]
> The name **Beta** suggests a _not really ready for production usage_ build and may annoy some people. Additionally we all know, that the released _Stable_ versions are always a bit buggy. 😉

I couldn't agree more.

I like the new naming scheme, @Dendraspis. Go on~ 😍

**Reply by stax76** [reply]
@Dendraspis 

It reads like a good plan, and it's really great to see that you have interest in such maintenance tasks.
